app/controllers/administrative-units/administrative-unit/change-events/new.js

```diff
@@ -65,55 +49,89 @@ export default class AdministrativeUnitsAdministrativeUnitChangeEventsNewControl
   }
 
   @action
-  validateEndDate(validation) {
-    this.endDateValidation = validateDate(validation);
-  }
-
-  @action
-  validateDate(validation) {
-    this.dateValidation = validateDate(validation);
+  filterSelectedOriginalOrganizations(searchResults) {
+    return searchResults.filter(
+      (organization) =>
+        !this.model.changeEvent.originalOrganizations.includes(organization)
+    );
   }
 
+  /**
+   * Update the original organizations related to the change event being
+   * created. If only one argument is provided this organization is added as a
+   * new original organization. Otherwise, if two arguments are provided the
+   * second one is considered as a replacement of the first. In other words the
+   * first organization will be removed as original organisation and the second
+   * organization is added as a new one.
+   * If the change event is being created for a central worship service, the
+   * provided arguments are also removed as possible resulting organizations.
+   * @param {AdministrativeUnitModel} previousOrganization
+   * @param {AdministrativeUnitModel} organization
+   */
   @action
-  validatePublicationEndDate(validation) {
-    this.publicationEndDateValidation = validateDate(validation);
+  updateOriginalOrganizations(previousOrganization, organization) {
+    if (organization) {
+      this.removeOriginalOrganization(previousOrganization);
+      this.model.changeEvent.addOriginalOrganization(organization);
+    } else {
```

Comment:
**Question**: I didn't find the where the method is called with 2 arguments. Is it still relevant?

Reply:
This function is called with two arguments when you change the selected organisation in a select-box that already had a value. The [template](https://github.com/lblod/frontend-organization-portal/blob/d9896e025a9b1f06ac2f786a419b33b3876616e9/app/templates/administrative-units/administrative-unit/change-events/new.hbs#L185) provides any already selected organisation as first argument.